In [16]:
!head /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3 -n 10

##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Myoluc2.0
#!genome-build-accession NCBI_Assembly:GCF_000147115.1
#!annotation-date 26 January 2018
#!annotation-source NCBI Myotis lucifugus Annotation Release 102
##sequence-region NW_005871048.1 1 64990205
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=59463
NW_005871048.1	RefSeq	region	1	64990205	.	+	.	ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female


In [ ]:
GL429891 266596

In [6]:
!wc -l /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf

482554 /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf


In [2]:
!head '/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNYVT40_margCompvalues.csv' -n 44

"","SNP","Scaffold","Maj","Min","NJ","NY","VT","Comp"
"15",282162,"GL429849","A","G",0.00111399888600111,0.00836899163100837,9.99999000001e-07,3.25319264647375e-09
"19",453486,"GL429872","A","G",0.000886999113000887,9.99999000001e-07,0.00143799856200144,5.14576123265913e-10
"20",463033,"GL429852","G","A",7.4999925000075e-05,0.000112999887000113,0.000434999565000435,1.37829106611689e-09
"21",463041,"GL429852","G","A",7.6999923000077e-05,1.999998000002e-06,0.00203599796400204,1.39290583823221e-10
"26",616121,"GL429834","G","C",9.99999000001e-07,0.000861999138000862,8.1999918000082e-05,3.46164465967714e-11
"51",1543734,"GL429882","T","C",0.00874799125200875,9.99999000001e-07,9.99999000001e-07,4.87504391237632e-12
"55",1735029,"GL429966","C","T",0.00301799698200302,0.00946499053500947,9.99999000001e-07,9.14116512034742e-09
"69",3296417,"GL429813","C","T",0.00786199213800786,9.99999000001e-07,0.00244999755000245,6.35734734029988e-09
"77",6378055,"GL429769","A","G",9.99999000001e-07,0.007934

Look at repeat regions (soft mask)

In [22]:
!head /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout2_sigCompvaluesNe424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","pvalues","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",51542832,"G","A",9.99999000001e-07,0.000253999746000254,5.86578523604657e-09,15,1.54259886773243e-07
"2","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000496999503000497,7.51112494690409e-07,81,8.33003388575513e-07
"3","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,4.999995000005e-06,6.76490539011496e-08,32,3.29087758449585e-07
"4","GL429767","NW_005871048.1",4550108,"T","C",0.00016999983000017,7.8999921000079e-05,2.56858669061094e-07,59,6.06755554641423e-07
"5","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,1.63656087014458e-09,10,1.02839924515495e-07
"6","GL429767","NW_005871048.1",38456248,"T","C",2.8999971000029e-05,0.00138399861600138,7.23690533813113e-07,80,8.22719396123963e-07
"7","GL429767","NW_005871048.1",51542790,"G","T",0.00101799898200102,3.999996000004e-06,8.27393463681677e-08,36,3.70223

In [ ]:
!wc -l ~/WGS/SNPS/All_mylu/snps/NJNY502_sigCompvalues_nr_Ne424_296.csv

In [46]:
from Bio import SeqIO

#tmp = open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout2_sigCompvaluesNe424_296.csv')
tmp = open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY502_sigCompvalues_nr_Ne424_296.csv')
lines = tmp.readlines()
lines[1]
tmp.close()
del tmp

In [47]:
scaffolds = {}

for line in lines[1:]:
    line = line.split(',')
    scaffold = line[1].replace('"','')
    index = int(line[3])
    
    if scaffold in scaffolds:
        scaffolds[scaffold].append(index)
    else:
        scaffolds[scaffold] = [index]

In [48]:
#o=open("/local/home/sarahgw/references/SNP_Scaffolds.fa")
read_data = []

for read in SeqIO.parse("/local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa", "fasta"):
    if read.id in scaffolds:
        read_data.append(read)
#print([read_data])

SeqIO.write(read_data, "/local/home/sarahgw/references/SNP_Scaffolds.fa", "fasta")
    
        

53

In [49]:
read_data = []

for read in SeqIO.parse("/local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa", "fasta"):
    if read.id in scaffolds:
        indexes = scaffolds[read.id]
        for idx in indexes:
            data = str(read.seq[idx-1])
            #data = str(read.seq[idx:idx+100])
            read_data.append([read.id, data, idx])
            print([read.id, data, idx])
        

['GL429767', 'T', 22649683]
['GL429767', 'A', 16839644]
['GL429767', 'C', 46534125]
['GL429768', 'A', 29962249]
['GL429768', 'A', 9041952]
['GL429768', 'G', 3921142]
['GL429770', 'A', 8507307]
['GL429772', 'T', 14993622]
['GL429773', 'T', 7204887]
['GL429775', 'A', 5556309]
['GL429778', 'G', 3561000]
['GL429781', 'G', 7594948]
['GL429783', 'G', 665407]
['GL429792', 'C', 753145]
['GL429801', 'T', 6986210]
['GL429802', 'C', 4091453]
['GL429806', 'C', 3311579]
['GL429816', 'G', 3229647]
['GL429819', 'G', 365354]
['GL429835', 'G', 2966433]
['GL429842', 'A', 4124241]
['GL429841', 'G', 612532]
['GL429841', 'G', 3522450]
['GL429845', 'T', 2091046]
['GL429855', 'G', 4401152]
['GL429859', 'T', 300331]
['GL429861', 'A', 3065399]
['GL429861', 'A', 293372]
['GL429867', 'G', 2575249]
['GL429873', 'T', 2056629]
['GL429879', 'G', 384336]
['GL429885', 'C', 617822]
['GL429885', 'T', 2043530]
['GL429885', 'T', 3445677]
['GL429888', 'A', 3508435]
['GL429910', 'C', 2414078]
['GL429927', 'A', 1241213]
['GL

In [50]:
count = 0
for read in read_data:
    #if read[1][0].islower():
    if read[1].islower():
        count += 1
print(count)
print(len(read_data))

0
62


In [37]:
o=open("/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_SNPs_repeat.txt", "w+")
for read in read_data:
    if read[1][0].islower():
        scaffold=read[0].replace('"','')
        snp=int(read[2])
        out = (read[0], snp)
        scafs= str(out)
        o.write(scafs+'\n')
        print scafs

('GL429767', 51542832)
('GL429767', 4550108)
('GL429767', 38456248)
('GL429767', 51542790)
('GL429770', 8292792)
('GL429784', 1229639)
('GL429795', 3907289)
('GL429828', 1501267)
('GL429835', 2966452)
('GL429839', 183883)
('GL429848', 2603829)
('GL429871', 64000)
('GL429896', 2002301)
('GL429906', 1866953)
('GL429938', 2500270)
('GL429947', 1200485)
('GL429969', 109273)
('GL429982', 1303809)
('GL430128', 735347)
('GL430132', 776142)
('GL430271', 326498)
('GL430834', 44020)
('GL430870', 25568)


In [1]:
!head /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_SNPs_repeat.txt

('GL429767', 51542832)
('GL429767', 4550108)
('GL429767', 38456248)
('GL429767', 51542790)
('GL429770', 8292792)
('GL429784', 1229639)
('GL429795', 3907289)
('GL429828', 1501267)
('GL429835', 2966452)
('GL429839', 183883)


In [10]:
for read in read_data:
    if read[1][0].islower():
        scaffold=read[0].replace('"','')
        snp=int(read[2])
        out = (read[0], snp, snp-300,snp+300)
        print out

('GL429767', 51542832, 51542532, 51543132)
('GL429767', 4550108, 4549808, 4550408)
('GL429767', 38456248, 38455948, 38456548)
('GL429767', 51542790, 51542490, 51543090)
('GL429770', 8292792, 8292492, 8293092)
('GL429784', 1229639, 1229339, 1229939)
('GL429795', 3907289, 3906989, 3907589)
('GL429828', 1501267, 1500967, 1501567)
('GL429835', 2966452, 2966152, 2966752)
('GL429839', 183883, 183583, 184183)
('GL429848', 2603829, 2603529, 2604129)
('GL429871', 64000, 63700, 64300)
('GL429896', 2002301, 2002001, 2002601)
('GL429906', 1866953, 1866653, 1867253)
('GL429938', 2500270, 2499970, 2500570)
('GL429947', 1200485, 1200185, 1200785)
('GL429969', 109273, 108973, 109573)
('GL429982', 1303809, 1303509, 1304109)
('GL430128', 735347, 735047, 735647)
('GL430132', 776142, 775842, 776442)
('GL430271', 326498, 326198, 326798)
('GL430834', 44020, 43720, 44320)
('GL430870', 25568, 25268, 25868)


In [ ]:
!cut -f1-2 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_sites.txt   |  awk '{print $1":"($2-300)"-"($2+300)}'| tr '\n' ' '

In [ ]:
samtools index /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam

In [ ]:
#samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam -r GL429767:4479268-4479868 GL429767:4549808-4550408 GL429767:16839344-16839944 GL429767:33884519-33885119 GL429767:46533825-46534425 GL429767:52446464-52447064 GL429768:446685-447285 GL429768:3920842-3921442 GL429768:9041652-9042252 GL429768:26061351-26061951 GL429768:34577493-34578093 GL429769:15245542-15246142 GL429770:1923576-1924176 GL429770:8292492-8293092 GL429770:10345356-10345956 GL429770:22531756-22532356 GL429770:24724952-24725552 GL429770:29523921-29524521 GL429771:3817117-3817717 GL429772:8715795-8716395 GL429772:14993322-14993922 GL429772:18592789-18593389 GL429773:12797725-12798325 GL429773:13172713-13173313 GL429774:3987738-3988338 GL429774:5007272-5007872 GL429775:6728203-6728803 GL429775:17037669-17038269 GL429776:3271309-3271909 GL429776:7915189-7915789 GL429776:15974471-15975071 GL429777:10443245-10443845 GL429777:10443250-10443850 GL429777:16214146-16214746 GL429778:11072905-11073505 GL429778:11429527-11430127 GL429778:12414329-12414929 GL429779:1937439-1938039 GL429779:3414312-3414912 GL429779:15275065-15275665 GL429781:7594648-7595248 GL429781:8738093-8738693 GL429781:12179326-12179926 GL429781:12179328-12179928 GL429781:12179338-12179938 GL429783:14617397-14617997 GL429784:1229339-1229939 GL429784:5836717-5837317 GL429785:3215849-3216449 GL429786:344044-344644 GL429789:4442217-4442817 GL429789:4442256-4442856 GL429791:4231908-4232508 GL429791:6065039-6065639 GL429791:6065358-6065958 GL429791:10731843-10732443 GL429794:8869775-8870375 GL429794:8869776-8870376 GL429797:8857437-8858037 GL429798:384987-385587 GL429802:4091153-4091753 GL429802:5027233-5027833 GL429803:5802459-5803059 GL429806:3311279-3311879 GL429810:3169341-3169941 GL429811:1437179-1437779 GL429811:6652943-6653543 GL429812:7410522-7411122 GL429813:1302893-1303493 GL429813:3296117-3296717 GL429813:4558705-4559305 GL429814:2311437-2312037 GL429815:936411-937011 GL429815:936422-937022 GL429818:830586-831186 GL429818:1105172-1105772 GL429819:4882431-4883031 GL429820:683980-684580 GL429820:6686985-6687585 GL429821:5481295-5481895 GL429821:6218554-6219154 GL429821:6730039-6730639 GL429824:6571401-6572001 GL429825:3483890-3484490 GL429827:2841476-2842076 GL429827:2985431-2986031 GL429828:1184669-1185269 GL429828:1500967-1501567 GL429828:2330239-2330839 GL429828:3004386-3004986 GL429830:84604-85204 GL429830:3004546-3005146 GL429831:2610188-2610788 GL429833:472530-473130 GL429835:2966130-2966730 GL429835:2966133-2966733 GL429835:2966152-2966752 GL429835:2966292-2966892 GL429836:943092-943692 GL429836:1939401-1940001 GL429836:2577955-2578555 GL429836:5267044-5267644 GL429838:708595-709195 GL429839:829466-830066 GL429840:5001597-5002197 GL429841:2186886-2187486 GL429842:129368-129968 GL429842:4123941-4124541 GL429845:2090746-2091346 GL429845:2090774-2091374 GL429846:4516669-4517269 GL429847:3937705-3938305 GL429854:1100834-1101434 GL429855:2585840-2586440 GL429855:4400852-4401452 GL429859:300031-300631 GL429860:112424-113024 GL429860:2304685-2305285 GL429862:3142551-3143151 GL429868:1361634-1362234 GL429871:63700-64300 GL429872:2510649-2511249 GL429873:2056329-2056929 GL429873:2585275-2585875 GL429877:135909-136509 GL429879:384036-384636 GL429879:2577313-2577913 GL429880:304800-305400 GL429883:1676909-1677509 GL429883:2387122-2387722 GL429884:1762675-1763275 GL429884:1883504-1884104 GL429885:3445377-3445977 GL429888:100154-100754 GL429889:639433-640033 GL429890:270093-270693 GL429890:1790243-1790843 GL429891:266296-266896 GL429893:1767354-1767954 GL429894:3396535-3397135 GL429896:1338399-1338999 GL429900:1012008-1012608 GL429900:3319780-3320380 GL429901:2731079-2731679 GL429906:1866653-1867253 GL429909:2495863-2496463 GL429910:1132409-1133009 GL429916:2268536-2269136 GL429920:2255391-2255991 GL429921:787524-788124 GL429921:1965190-1965790 GL429927:507243-507843 GL429927:507244-507844 GL429929:2602094-2602694 GL429930:2495100-2495700 GL429930:2585979-2586579 GL429933:442672-443272 GL429938:2499970-2500570 GL429940:1631828-1632428 GL429942:2344778-2345378 GL429946:160670-161270 GL429950:1108250-1108850 GL429955:358993-359593 GL429955:1302398-1302998 GL429955:1478137-1478737 GL429955:1484286-1484886 GL429955:2107567-2108167 GL429956:1589704-1590304 GL429958:1185328-1185928 GL429960:210893-211493 GL429960:2144369-2144969 GL429961:677230-677830 GL429962:688993-689593 GL429962:1318791-1319391 GL429964:242275-242875 GL429964:1628637-1629237 GL429965:1294619-1295219 GL429965:1956678-1957278 GL429966:914505-915105 GL429968:2123597-2124197 GL429969:375069-375669 GL429977:2174455-2175055 GL429984:2098221-2098821 GL429987:78505-79105 GL429990:1916625-1917225 GL429996:1834660-1835260 GL429998:99024-99624 GL430004:183065-183665 GL430008:1395810-1396410 GL430012:1445576-1446176 GL430016:1091288-1091888 GL430018:179978-180578 GL430019:347245-347845 GL430026:1213403-1214003 GL430027:535774-536374 GL430032:318118-318718 GL430034:1478380-1478980 GL430036:626124-626724 GL430040:260314-260914 GL430042:401057-401657 GL430044:232993-233593 GL430045:1217420-1218020 GL430048:360599-361199 GL430048:873994-874594 GL430053:149163-149763 GL430055:424788-425388 GL430058:488672-489272 GL430074:994574-995174 GL430121:936284-936884 GL430128:735047-735647 GL430151:194628-195228 GL430155:309594-310194 GL430156:916361-916961 GL430158:593541-594141 GL430174:463636-464236 GL430187:529350-529950 GL430221:641434-642034 GL430234:695806-696406 GL430239:60025-60625 GL430239:60026-60626 GL430265:592702-593302 GL430281:513767-514367 GL430282:487760-488360 GL430285:514771-515371 GL430287:351710-352310 GL430319:176873-177473 GL430320:360802-361402 GL430321:292319-292919 GL430328:469941-470541 GL430382:291250-291850 GL430403:54529-55129 GL430420:180354-180954 GL430429:127606-128206 GL430469:116218-116818 GL430482:173052-173652 GL430565:94659-95259 GL430577:70583-71183 GL430746:9363-9963 GL430898:10806-11406 GL431215:29414-30014 GL432768:665-1265 AAPE02064211:5952-6552 AAPE02069524:2281-2881 AAPE02069524:2285-2885 AAPE02069524:2286-2886 GL429767:4379847-4380447 GL429767:22649383-22649983 GL429767:38455948-38456548 GL429767:51542490-51543090 GL429767:51542532-51543132 GL429767:59532782-59533382 GL429768:29961949-29962549 GL429769:16777537-16778137 GL429769:26665511-26666111 GL429770:8507007-8507607 GL429770:16029084-16029684 GL429770:16029098-16029698 GL429772:4817046-4817646 GL429772:12062966-12063566 GL429772:17021875-17022475 GL429772:17021882-17022482 GL429772:17021892-17022492 GL429773:116772-117372 GL429773:284768-285368 GL429773:1036637-1037237 GL429773:7204587-7205187 GL429774:5086619-5087219 GL429775:1710476-1711076 GL429775:5556009-5556609 GL429775:10613987-10614587 GL429775:17830735-17831335 GL429776:3462655-3463255 GL429776:13622792-13623392 GL429776:16056773-16057373 GL429778:3560700-3561300 GL429778:6871768-6872368 GL429778:13606897-13607497 GL429779:2091418-2092018 GL429779:2407488-2408088 GL429779:10152144-10152744 GL429782:13104338-13104938 GL429782:13104343-13104943 GL429782:13679850-13680450 GL429783:665107-665707 GL429783:2691560-2692160 GL429783:2782429-2783029 GL429783:13929195-13929795 GL429784:440583-441183 GL429784:4614537-4615137 GL429784:10091192-10091792 GL429785:5038507-5039107 GL429785:5735592-5736192 GL429785:6643107-6643707 GL429785:7041667-7042267 GL429787:11158263-11158863 GL429789:12124990-12125590 GL429789:12124992-12125592 GL429789:12124993-12125593 GL429790:1037818-1038418 GL429790:7965198-7965798 GL429790:10304028-10304628 GL429792:752845-753445 GL429793:8414502-8415102 GL429793:8414508-8415108 GL429794:4072396-4072996 GL429794:5707570-5708170 GL429794:7279602-7280202 GL429794:8583957-8584557 GL429795:3906989-3907589 GL429797:6341030-6341630 GL429798:6594618-6595218 GL429798:9304648-9305248 GL429800:5353206-5353806 GL429801:6985910-6986510 GL429801:6985920-6986520 GL429803:8026244-8026844 GL429806:2488686-2489286 GL429808:770277-770877 GL429809:6286974-6287574 GL429809:7389734-7390334 GL429810:2353609-2354209 GL429811:2139684-2140284 GL429815:1149355-1149955 GL429816:3229347-3229947 GL429819:365054-365654 GL429824:3462714-3463314 GL429824:4399436-4400036 GL429826:127215-127815 GL429829:5660877-5661477 GL429831:1081416-1082016 GL429831:5178277-5178877 GL429836:943173-943773 GL429836:1894528-1895128 GL429836:2323268-2323868 GL429839:183583-184183 GL429839:2333752-2334352 GL429841:612232-612832 GL429841:3522150-3522750 GL429845:2824589-2825189 GL429845:4476103-4476703 GL429846:940795-941395 GL429846:1252876-1253476 GL429846:2542039-2542639 GL429846:4607696-4608296 GL429848:2603529-2604129 GL429849:815477-816077 GL429850:1182854-1183454 GL429850:2489681-2490281 GL429851:842283-842883 GL429852:462733-463333 GL429852:462741-463341 GL429852:462802-463402 GL429854:64393-64993 GL429857:134312-134912 GL429860:863225-863825 GL429860:3647355-3647955 GL429861:293072-293672 GL429861:3065099-3065699 GL429861:4317217-4317817 GL429861:4317223-4317823 GL429862:268802-269402 GL429863:2978300-2978900 GL429863:4280244-4280844 GL429865:2201021-2201621 GL429867:2184951-2185551 GL429867:2574949-2575549 GL429868:2981897-2982497 GL429870:2909520-2910120 GL429871:2083045-2083645 GL429874:3960541-3961141 GL429880:42524-43124 GL429880:1266267-1266867 GL429881:677125-677725 GL429881:1997805-1998405 GL429882:2695846-2696446 GL429883:509119-509719 GL429884:1762764-1763364 GL429884:1762785-1763385 GL429884:1803449-1804049 GL429885:617522-618122 GL429885:617526-618126 GL429885:2043230-2043830 GL429886:1745769-1746369 GL429887:2687102-2687702 GL429887:2687123-2687723 GL429888:3508135-3508735 GL429896:2002001-2002601 GL429896:2866533-2867133 GL429898:676397-676997 GL429898:676414-677014 GL429899:351544-352144 GL429904:1874105-1874705 GL429905:930988-931588 GL429905:1245900-1246500 GL429910:2413778-2414378 GL429910:2565017-2565617 GL429917:581316-581916 GL429918:2724708-2725308 GL429925:661983-662583 GL429925:2311366-2311966 GL429927:1240913-1241513 GL429928:2425225-2425825 GL429929:2386002-2386602 GL429929:2943633-2944233 GL429932:112161-112761 GL429939:2758936-2759536 GL429944:757168-757768 GL429946:1038779-1039379 GL429947:1200185-1200785 GL429949:1338890-1339490 GL429949:1953162-1953762 GL429950:2444579-2445179 GL429952:330669-331269 GL429953:2190150-2190750 GL429959:771290-771890 GL429964:1805499-1806099 GL429969:108973-109573 GL429979:2240502-2241102 GL429982:1303509-1304109 GL429995:282544-283144 GL429995:731342-731942 GL429995:846763-847363 GL430004:1824263-1824863 GL430007:1175078-1175678 GL430008:1354902-1355502 GL430015:641111-641711 GL430022:533572-534172 GL430029:1661113-1661713 GL430054:1392788-1393388 GL430063:1016282-1016882 GL430071:327435-328035 GL430088:571173-571773 GL430101:496608-497208 GL430124:395553-396153 GL430132:775842-776442 GL430133:495971-496571 GL430133:757020-757620 GL430144:296011-296611 GL430144:350723-351323 GL430145:65372-65972 GL430162:78373-78973 GL430164:554639-555239 GL430166:11632-12232 GL430199:554097-554697 GL430213:519259-519859 GL430232:111038-111638 GL430234:78289-78889 GL430271:326198-326798 GL430271:524910-525510 GL430279:606865-607465 GL430338:350247-350847 GL430384:251940-252540 GL430403:269522-270122 GL430441:181204-181804 GL430483:252824-253424 GL430551:168312-168912 GL430678:36391-36991 GL430686:90734-91334 GL430696:55139-55739 GL430834:43720-44320 GL430870:25268-25868 GL431096:35116-35716 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bam

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429767:51542532-51543132  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_51542832.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_51542832.bam

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429767:4549808-4550408  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_4550108.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_4550108.bam

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429767:38455948-38456548  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_38456248.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_38456248.bam    

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429767:51542490-51543090  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_51542790.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_51542790.bam    

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429770:8292492-8293092  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_8292792.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_8292792.bam    

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429784:1229339-1229939  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1229639.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1229639.bam    

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429795:3906989-3907589  > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_3907589.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_3907589.bam    

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429828:1500967-1501567 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1501267.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1501267.bam  

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429835:2966152-2966752 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_2966452.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_2966452.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429839:183583-184183 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_183883.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_183883.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429871:63700-64300 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_64000.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_64000.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429906:1866653-1867253 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1866953.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1866953.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429906:2499970-2500570 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_2500270.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_2500270.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429947:1200185-1200785 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1200485.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1200485.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429969:108973-109573 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_109273.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_109273.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL429982:1303509-1304109 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1303809.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_1303809.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL430128:735047-735647 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_735347.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_735347.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL430132:775842-776442 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_776142.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_776142.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL430271:326198-326798 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_326498.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_326498.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL430834:43720-44320 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_44020.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_44020.bam 

In [ ]:
samtools view -b /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam GL430870:25268-25868 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_25568.bam
samtools index /local/home/sarahgw/WGS/SNPS/All_mylu/snps/SNP_bam_files/njnyeu50_600_25568.bam 

In [6]:
!cut -f1-2 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_sites.txt   |  awk -v OFS='\t' '{print $1, ($2-300), ($2+300)}'> /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bed 

In [10]:
!tail -n+2 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bed > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600h.bed

In [7]:
print "chrom\tchromStart\tchromEnd"

chrom	chromStart	chromEnd


In [11]:
!echo -e "chrom\tchromStart\tchromEnd" | cat - /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600h.bed > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bed

In [12]:
!head /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bed 

chrom	chromStart	chromEnd
GL429767	4479268	4479868
GL429767	4549808	4550408
GL429767	16839344	16839944
GL429767	33884519	33885119
GL429767	46533825	46534425
GL429767	52446464	52447064
GL429768	446685	447285
GL429768	3920842	3921442
GL429768	9041652	9042252


GL429767	4479268	4479868
GL429767	4549808	4550408
GL429767	16839344	16839944
GL429767	33884519	33885119
GL429767	46533825	46534425
GL429767	52446464	52447064
GL429768	446685	447285
GL429768	3920842	3921442
GL429768	9041652	9042252
GL429768	26061351	26061951
GL429768	34577493	34578093
GL429769	15245542	15246142
GL429770	1923576	1924176
GL429770	8292492	8293092
GL429770	10345356	10345956
GL429770	22531756	22532356
GL429770	24724952	24725552
GL429770	29523921	29524521
GL429771	3817117	3817717
GL429772	8715795	8716395
GL429772	14993322	14993922
GL429772	18592789	18593389
GL429773	12797725	12798325
GL429773	13172713	13173313
GL429774	3987738	3988338
GL429774	5007272	5007872
GL429775	6728203	6728803
GL429775	17037669	17038269
GL429776	3271309	3271909
GL429776	7915189	7915789
GL429776	15974471	15975071
GL429777	10443245	10443845
GL429777	10443250	10443850
GL429777	16214146	16214746
GL429778	11072905	11073505
GL429778	11429527	11430127
GL429778	12414329	12414929
GL429779	1937439	1938039
GL4297

In [ ]:
samtools view -b -L /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bed /local/home/sarahgw/WGS/Sequences/All_mylu/All_Mylu_bam/All_mylu_merged.bam > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/njnyeu50_600.bam

In [21]:
!head /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.92.gtf

#!genome-build Myoluc2.0
#!genome-version Myoluc2.0
#!genome-date 2010-09
#!genome-build-accession NCBI:GCA_000147115.1
#!genebuild-last-updated 2011-06
GL429767	ensembl	gene	42604	226589	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding";
GL429767	ensembl	transcript	42604	226589	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; transcript_id "ENSMLUT00000025902"; transcript_version "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ACHE-201"; transcript_source "ensembl"; transcript_biotype "protein_coding";
GL429767	ensembl	exon	42604	43700	.	+	.	gene_id "ENSMLUG00000029214"; gene_version "1"; transcript_id "ENSMLUT00000025902"; transcript_version "1"; exon_number "1"; gene_name "ACHE"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ACHE-201"; transcript_source "ensembl"; transcript_biotype "protein_coding"; exon_id "ENSMLUE00000285193"; exon

In [22]:
!head -n 92 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","NY","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000181999818000182,2.90413319116625e-07,70,5.39909603706351e-07
"2","GL429767","NW_005871048.1",38456248,"T","C",2.8999971000029e-05,0.000823999176000824,4.43259321723044e-07,78,6.01613558415648e-07
"3","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,9.99999000001e-07,1.45904282432881e-08,20,1.54259886773243e-07
"4","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,1.63656087014458e-09,9,6.94169490479594e-08
"5","GL429767","NW_005871048.1",51542832,"G","A",9.99999000001e-07,0.000233999766000234,5.42310342192109e-09,13,1.00268926402608e-07
"6","GL429768","NW_005871049.1",3921142,"G","C",9.99999000001e-07,0.0138399861600138,2.64284027064383e-07,67,5.16770620690364e-07
"7","GL429768","NW_005871049.1",34577793,"C","T",0.000584999415000585,2.3999976000024e-05,2.6790172263069e-07,68,5.2448361502902

In [ ]:
2,125,561-2,462,498   2,603,829

In [ ]:
46,836,518..47,675,316

In [ ]:
2,123,766, 3,029,105

MASP1- A
GALNT18- A
BMPER
GNPDA1
BFSP1
REPS2 (REPS1 A)
PCDH17 ( PCDH15 A PCDH11)
SOX5-A
MDGA2-A
CACNA1D
FA2H
TPRA1
ADCY3- A
CEP112
TG -A 
NELL1 - A
SEMA5A -A
SORCS3 - A
CBFA2T3 SHOULD 
SGCZ -A 
GEMIN4 (GEMIN7 A)
MKL2 
BACE2
RAPGEF1 -A 

There are 1,591 genes in that set...can we see how many are in the genome, and see how many genes are in our genome. 

can we do something like epstein We sampled 1,000 pairs of 200 kb regions around randomly selected SNPs in our data set, and only 25% of such regions contained one or more genes with Gene Ontology14 terms ‘immune response,’ ‘cell proliferation,’ or ‘cell death’ (our candidate regions contained both an immune response and a cell death GO annotation). Consequently, our result (two out of seven based on GO terms or five out of seven based on manual annotation) is unlikely to have been found by chance.




MASP1
GALNT18
BMPER
GNPDA1
BFSP1
REPS2
PCDH17
SOX5
MDGA2
CACNA1D- GNRH, INSULIN
FA2H
TPRA1
ADCY3- GNRH, INSULIN
CEP112
TG
NELL1
SEMA5A
SORCS3
CBFA2T3
SGCZ
GEMIN4
MKL2
BACE2
RAPGEF1- RAPGEF4 INSULIN


In [15]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout2_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    if args[2] in chrom_dict:
        chrom_dict[args[2]].append(int(args[3]))
    else:
        chrom_dict[args[2]] = [int(args[3])]
        
aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        for loc in chrom_dict[chrom]:
        # loc = chrom_dict[chrom]
            if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
                count += 1
                print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chrom

['NW_005871087.1', 'RefSeq', 'region', '1', '8424078', '.', '+', '.', 'ID=id142776;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871097.1', 'RefSeq', 'region', '1', '7348857', '.', '+', '.', 'ID=id162626;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871097.1', 'Gnomon', 'gene', '3142067', '3321329', '.', '-', '.', 'ID=gene8121;Dbxref=GeneID:102440988;Name=REPS2;gbkey=Gene;gene=REPS2;gene_biotype=protein_coding\n']
['NW_005871097.1', 'Gnomon', 'mRNA', '3142067', '3321329', '.', '-', '.', 'ID=rna13665;Parent=gene8121;Dbxref=GeneID:102440988,Genbank:XM_023753740.1;Name=XM_023753740.1;gbkey=mRNA;gene=REPS2;model_evidence=Supporting evidence includes similarity to: 1 Protein%2C and 82%25 coverage of the annotated genomic feature by RNAseq alignments;product=RALBP1 associated Eps domain 

['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_006090529.3;Name=XM_006090529.3;gbkey=mRNA;gene=PCDH17;model_evidence=Supporting evidence includes similarity to: 2 mRNAs%2C 5 Proteins%2C and 95%25 coverage of the annotated genomic feature by RNAseq alignments;product=protocadheri

['NW_005871219.1', 'RefSeq', 'region', '1', '2865968', '.', '+', '.', 'ID=id315315;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871228.1', 'RefSeq', 'region', '1', '2723584', '.', '+', '.', 'ID=id320747;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871231.1', 'RefSeq', 'region', '1', '2651257', '.', '+', '.', 'ID=id323591;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871236.1', 'RefSeq', 'region', '1', '2551290', '.', '+', '.', 'ID=id327813;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871243.1', 'RefSeq', 'region', '1', '2513712', '.', '+', '.', 'ID=id331702;Dbxref=taxon:5946

['NW_005871763.1', 'RefSeq', 'region', '1', '283313', '.', '+', '.', 'ID=id502584;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871846.1', 'RefSeq', 'region', '1', '211799', '.', '+', '.', 'ID=id512757;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:1024175

160

In [6]:
!head -n 62 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY502_sigCompvalues_nr_Ne424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","NY","NY_out","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,9.99999000001e-07,4.999995000005e-06,6.76490539011496e-08,22,3.22067863065821e-07
"2","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000181999818000182,0.000496999503000497,7.51112494690409e-07,58,8.49088002628074e-07
"3","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,9.99999000001e-07,1.63656087014458e-09,7,1.02476138248216e-07
"4","GL429768","NW_005871049.1",29962249,"A","G",9.99999000001e-07,0.00581299418700581,0.00524199475800524,1.05188721947827e-07,26,3.80625656350516e-07
"5","GL429768","NW_005871049.1",9041952,"A","G",0.048968951031049,1.5999984000016e-05,9.99999000001e-07,8.73217398285339e-07,61,8.93006347591595e-07
"6","GL429768","NW_005871049.1",3921142,"G","C",9.99999000001e-07,0.0138399861600138,0.0280749719250281,5.16252970035752e-07,50,7.31972416058684e-07
"7","GL42

In [5]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY502_sigCompvalues_nr_Ne424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    if args[2] in chrom_dict:
        chrom_dict[args[2]].append(int(args[3]))
    else:
        chrom_dict[args[2]] = [int(args[3])]
        
aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        for loc in chrom_dict[chrom]:
        # loc = chrom_dict[chrom]
            if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
                count += 1
                print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871048.1', 'Gnomon', 'gene', '16490629', '16935896', '.', '+', '.', 'ID=gene225;Dbxref=GeneID:102417759;Name=FBXL17;gbkey=Gene;gene=FBXL17;gene_biotype=protein_coding\n']
['NW_005871048.1', 'Gnomon', 'mRNA', '16490629', '16935896', '.', '+', '.', 'ID=rna402;Parent=gene225;Dbxref=GeneID:102417759,Genbank:XM_006081173.2;Name=XM_0

['NW_005871116.1', 'RefSeq', 'region', '1', '5728823', '.', '+', '.', 'ID=id192215;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_0060905

['NW_005871236.1', 'RefSeq', 'region', '1', '2551290', '.', '+', '.', 'ID=id327813;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871243.1', 'RefSeq', 'region', '1', '2513712', '.', '+', '.', 'ID=id331702;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871246.1', 'RefSeq', 'region', '1', '2532579', '.', '+', '.', 'ID=id335396;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871246.1', 'Gnomon', 'gene', '1936775', '1992192', '.', '-', '.', 'ID=gene16883;Dbxref=GeneID:102429022;Name=CBFA2T3;gbkey=Gene;gene=CBFA2T3;gene_biotype=protein_coding\n']
['NW_005871246.1', 'Gnomon', 'mRNA', '1936775', '1992192', '.', '-', '.', 'ID=rna29149;Parent=gene16883;Dbxref=GeneID:102429022,Genbank:XM_00609

['NW_005871763.1', 'RefSeq', 'region', '1', '283313', '.', '+', '.', 'ID=id502584;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871846.1', 'RefSeq', 'region', '1', '211799', '.', '+', '.', 'ID=id512757;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:1024175

105

Myso specific SNPs

In [1]:
f=open('/local/home/sarahgw/WGS/SNPS/Myso/mafs/Myso_SNPs_high.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    if args[2] in chrom_dict:
        chrom_dict[args[2]].append(int(args[3]))
    else:
        chrom_dict[args[2]] = [int(args[3])]
        
aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        for loc in chrom_dict[chrom]:
        # loc = chrom_dict[chrom]
            if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
                count += 1
                print agroups
    aline = a.readline()
count

['NW_005871062.1', 'RefSeq', 'region', '1', '15274826', '.', '+', '.', 'ID=id85264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871062.1', 'Gnomon', 'gene', '7989018', '8014069', '.', '-', '.', 'ID=gene4100;Dbxref=GeneID:102441373;Name=LOC102441373;gbkey=Gene;gene=LOC102441373;gene_biotype=protein_coding\n']
['NW_005871062.1', 'Gnomon', 'mRNA', '7989018', '8014069', '.', '-', '.', 'ID=rna7076;Parent=gene4100;Dbxref=GeneID:102441373,Genbank:XM_014452242.2;Name=XM_014452242.2;gbkey=mRNA;gene=LOC102441373;model_evidence=Supporting evidence includes similarity to: 1 Protein%2C and 73%25 coverage of the annotated genomic feature by RNAseq alignments;product=kininogen-1-like;transcript_id=XM_014452242.2\n']
['NW_005871070.1', 'RefSeq', 'region', '1', '12132627', '.', '+', '.', 'ID=id108328;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=ge

['NW_005871655.1', 'RefSeq', 'region', '1', '417845', '.', '+', '.', 'ID=id487368;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871655.1', 'RefSeq', 'region', '1', '417845', '.', '+', '.', 'ID=id487368;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871671.1', 'RefSeq', 'region', '1', '461733', '.', '+', '.', 'ID=id490022;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871718.1', 'RefSeq', 'region', '1', '444311', '.', '+', '.', 'ID=id496604;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871763.1', 'RefSeq', 'region', '1', '283313', '.', '+', '.', 'ID=id502584;Dbxref=taxon:59463;Nam

78

In [3]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50noout1.5_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    chrom_dict[args[2]] = int(args[3])

aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        loc = chrom_dict[chrom]
        if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
            count += 1
            print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871049.1', 'RefSeq', 'region', '1', '39494369', '.', '+', '.', 'ID=id14505;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871051.1', 'RefSeq', 'region', '1', '30916457', '.', '+', '.', 'ID=id28999;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'RefSeq', 'region', '1', '20945196', '.', '+', '.', 'ID=id44306;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'Gnomon', 'gene', '8711696', '8718521', '.', '+', '.', 'ID=gene2288;Dbxref=GeneID:111

['NW_005871122.1', 'RefSeq', 'region', '1', '5136827', '.', '+', '.', 'ID=id201139;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871122.1', 'Gnomon', 'gene', '3434412', '3529751', '.', '-', '.', 'ID=gene10137;Dbxref=GeneID:102437398;Name=PCDH17;gbkey=Gene;gene=PCDH17;gene_biotype=protein_coding\n']
['NW_005871122.1', 'Gnomon', 'mRNA', '3434412', '3529751', '.', '-', '.', 'ID=rna17052;Parent=gene10137;Dbxref=GeneID:102437398,Genbank:XM_006090529.3;Name=XM_006090529.3;gbkey=mRNA;gene=PCDH17;model_evidence=Supporting evidence includes similarity to: 2 mRNAs%2C 5 Proteins%2C and 95%25 coverage of the annotated genomic feature by RNAseq alignments;product=protocadherin 17;transcript_id=XM_006090529.3\n']
['NW_005871123.1', 'RefSeq', 'region', '1', '5160323', '.', '+', '.', 'ID=id201293;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomi

['NW_005871219.1', 'RefSeq', 'region', '1', '2865968', '.', '+', '.', 'ID=id315315;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871228.1', 'RefSeq', 'region', '1', '2723584', '.', '+', '.', 'ID=id320747;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871236.1', 'RefSeq', 'region', '1', '2551290', '.', '+', '.', 'ID=id327813;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871243.1', 'RefSeq', 'region', '1', '2513712', '.', '+', '.', 'ID=id331702;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871246.1', 'RefSeq', 'region', '1', '2532579', '.', '+', '.', 'ID=id335396;Dbxref=taxon:5946

['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:102417583;Name=RAPGEF1;gbkey=Gene;gene=RAPGEF1;gene_biotype=protein_coding\n']
['NW_005872028.1', 'Gnomon', 'mRNA', '5481', '98250', '.', '-', '.', 'ID=rna46592;Parent=gene27256;Dbxref=GeneID:102417583,Genbank:XM_023747795.1;Name=XM_023747795.1;gbkey=mRNA;gene=RAPGEF1;model_evidence=Supporting evidence includes similarity to: 7 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 1 sample with support for 

115

In [9]:
f=open('/local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv', 'r')
a=open('/local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3', 'r')
flines = f.readlines()[1:]
chrom_dict = {}
for line in flines:
    line = line.replace('"', '')
    args = line.split(',')
    chrom_dict[args[2]] = int(args[3])

aline = a.readline()
count = 0
while aline:
    agroups = aline.split('\t')
    #print(agroups)
    chrom = agroups[0]
    if chrom in chrom_dict:
        loc = chrom_dict[chrom]
        if int(agroups[3]) < (loc) and int(agroups[4]) > (loc):
            count += 1
            print agroups
    aline = a.readline()
count

['NW_005871048.1', 'RefSeq', 'region', '1', '64990205', '.', '+', '.', 'ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871049.1', 'RefSeq', 'region', '1', '39494369', '.', '+', '.', 'ID=id14505;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871051.1', 'RefSeq', 'region', '1', '30916457', '.', '+', '.', 'ID=id28999;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'RefSeq', 'region', '1', '20945196', '.', '+', '.', 'ID=id44306;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871053.1', 'Gnomon', 'gene', '14979316', '15072600', '.', '-', '.', 'ID=gene2361;Dbxref=GeneID:1

['NW_005871109.1', 'RefSeq', 'region', '1', '6495742', '.', '+', '.', 'ID=id181624;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871109.1', 'Gnomon', 'gene', '2888188', '3176295', '.', '+', '.', 'ID=gene9224;Dbxref=GeneID:102435828;Name=GALNT18;gbkey=Gene;gene=GALNT18;gene_biotype=protein_coding\n']
['NW_005871109.1', 'Gnomon', 'mRNA', '2888188', '3176295', '.', '+', '.', 'ID=rna15475;Parent=gene9224;Dbxref=GeneID:102435828,Genbank:XM_006089610.3;Name=XM_006089610.3;gbkey=mRNA;gene=GALNT18;model_evidence=Supporting evidence includes similarity to: 6 mRNAs%2C 7 Proteins%2C and 99%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 9 samples with support for all annotated introns;product=polypeptide N-acetylgalactosaminyltransferase 18%2C transcript variant X1;transcript_id=XM_006089610.3\n']
['NW_005871109.1', 'Gnomon', 'mRNA', '2888188', '3176292', '.', '+'

['NW_005871148.1', 'RefSeq', 'region', '1', '4444721', '.', '+', '.', 'ID=id238203;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871148.1', 'Gnomon', 'gene', '2560994', '2842740', '.', '-', '.', 'ID=gene11898;Dbxref=GeneID:102438113;Name=CEP112;gbkey=Gene;gene=CEP112;gene_biotype=protein_coding\n']
['NW_005871148.1', 'Gnomon', 'mRNA', '2560994', '2842740', '.', '-', '.', 'ID=rna20342;Parent=gene11898;Dbxref=GeneID:102438113,Genbank:XM_014458887.2;Name=XM_014458887.2;gbkey=mRNA;gene=CEP112;model_evidence=Supporting evidence includes similarity to: 2 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments;product=centrosomal protein 112;transcript_id=XM_014458887.2\n']
['NW_005871154.1', 'RefSeq', 'region', '1', '4280553', '.', '+', '.', 'ID=id250845;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;m

['NW_005871263.1', 'RefSeq', 'region', '1', '2259905', '.', '+', '.', 'ID=id349927;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871279.1', 'RefSeq', 'region', '1', '1996049', '.', '+', '.', 'ID=id357671;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871289.1', 'RefSeq', 'region', '1', '1901218', '.', '+', '.', 'ID=id361809;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871299.1', 'RefSeq', 'region', '1', '1811804', '.', '+', '.', 'ID=id366663;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871310.1', 'RefSeq', 'region', '1', '1841132', '.', '+', '.', 'ID=id371601;Dbxref=taxon:5946

['NW_005871967.1', 'RefSeq', 'region', '1', '145539', '.', '+', '.', 'ID=id520168;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005871977.1', 'RefSeq', 'region', '1', '146487', '.', '+', '.', 'ID=id520514;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'RefSeq', 'region', '1', '123542', '.', '+', '.', 'ID=id524264;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female\n']
['NW_005872028.1', 'Gnomon', 'gene', '5481', '98250', '.', '-', '.', 'ID=gene27256;Dbxref=GeneID:102417583;Name=RAPGEF1;gbkey=Gene;gene=RAPGEF1;gene_biotype=protein_coding\n']
['NW_005872028.1', 'Gnomon', 'mRNA', '5481', '98250', '.', '-', '.', 'ID=rna46592;Parent=gene27256;Dbxref=GeneID:102417583,Genbank:XM_023747795.1;Name=X

145

In [ ]:
p vals 

2603829 MDGA2,  in our sig all 3
616121 unknown, but in our sig all 3
35761989 unknown, but in our sig all 3

## VEP for synonymous/non synonymous

create input file of sig snps

The default format is a simple whitespace-separated format (columns may be separated by space or tab characters), containing five required columns plus an optional identifier column:

chromosome - just the name or number, with no 'chr' prefix
start
end
allele - pair of alleles separated by a '/', with the reference allele first
strand - defined as + (forward) or - (reverse).
identifier - this identifier will be used in VEP's output. If not provided, VEP will construct an identifier from the given coordinates and alleles.

In [18]:
!head /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv

"","Scaffold","NewScaf","SNP","Maj","Min","NJ","NY","Comp","rank","padjBH"
"1","GL429767","NW_005871048.1",16839644,"T","A",8.3999916000084e-05,0.000181999818000182,2.90413319116625e-07,70,5.39909603706351e-07
"2","GL429767","NW_005871048.1",38456248,"T","C",2.8999971000029e-05,0.000823999176000824,4.43259321723044e-07,78,6.01613558415648e-07
"3","GL429767","NW_005871048.1",22649683,"T","A",0.000658999341000659,9.99999000001e-07,1.45904282432881e-08,20,1.54259886773243e-07
"4","GL429767","NW_005871048.1",46534125,"C","T",6.6999933000067e-05,9.99999000001e-07,1.63656087014458e-09,9,6.94169490479594e-08
"5","GL429767","NW_005871048.1",51542832,"G","A",9.99999000001e-07,0.000233999766000234,5.42310342192109e-09,13,1.00268926402608e-07
"6","GL429768","NW_005871049.1",3921142,"G","C",9.99999000001e-07,0.0138399861600138,2.64284027064383e-07,67,5.16770620690364e-07
"7","GL429768","NW_005871049.1",34577793,"C","T",0.000584999415000585,2.3999976000024e-05,2.6790172263069e-07,68,5.2448361502902

In [26]:
!awk 'BEGIN { FS = "," }{print $2,$4,$4,$5"/"$6, "+"}' /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv|sed 's/\"//g'| tail -n 92 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP.txt    

In [28]:
!awk 'BEGIN { FS = "," }{print $3,$4,$4,$5"/"$6, "+"}' /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296.csv|sed 's/\"//g'| tail -n 92 > /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt    

In [ ]:
/local/home/sarahgw/ensembl-vep/vep -i /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP.txt --cache --species myotis_lucifugus 

bgzip and tabix file

In [23]:
!head /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3

##gff-version 3
#!gff-spec-version 1.21
#!processor NCBI annotwriter
#!genome-build Myoluc2.0
#!genome-build-accession NCBI_Assembly:GCF_000147115.1
#!annotation-date 26 January 2018
#!annotation-source NCBI Myotis lucifugus Annotation Release 102
##sequence-region NW_005871048.1 1 64990205
##species https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=59463
NW_005871048.1	RefSeq	region	1	64990205	.	+	.	ID=id0;Dbxref=taxon:59463;Name=Unknown;chromosome=Unknown;country=USA: Framingham%2C MA;gbkey=Src;genome=genomic;mol_type=genomic DNA;sex=female


In [29]:
!grep -v "#" /local/home/sarahgw/references/ref_Myoluc2.0_top_level.gff3 |  sort -k1,1 -k4,4n > /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3

In [ ]:
bgzip -c /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3 > /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz
tabix -p gff /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz

In [ ]:
bgzip -c /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa > /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa.gz

In [30]:
/local/home/sarahgw/ensembl-vep/vep  -i /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt -fasta /local/home/sarahgw/references/Myotis_lucifugus.Myoluc2.0.dna_sm.toplevel.fa --gff /local/home/sarahgw/references/ref_Myoluc2.0_top_level_sorted.gff3.gz --cache --species myotis_lucifugus 


SyntaxError: invalid syntax (<ipython-input-30-f0842e4a7489>, line 1)

In [1]:
!head -n 100 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/NJNY50_sigCompvaluesNe424_296_inputVEP2.txt

NW_005871048.1 16839644 16839644 T/A +
NW_005871048.1 38456248 38456248 T/C +
NW_005871048.1 22649683 22649683 T/A +
NW_005871048.1 46534125 46534125 C/T +
NW_005871048.1 51542832 51542832 G/A +
NW_005871049.1 3921142 3921142 G/C +
NW_005871049.1 34577793 34577793 C/T +
NW_005871049.1 29962249 29962249 A/G +
NW_005871051.1 8292792 8292792 A/T +
NW_005871051.1 8507307 8507307 A/C +
NW_005871053.1 8716095 8716095 T/A +
NW_005871053.1 14993622 14993622 T/C +
NW_005871054.1 7204887 7204887 C/T +
NW_005871056.1 5556309 5556309 A/C +
NW_005871059.1 3561000 3561000 G/A +
NW_005871062.1 7594948 7594948 A/T +
NW_005871064.1 14617697 14617697 C/A +
NW_005871064.1 665407 665407 C/T +
NW_005871065.1 1229639 1229639 G/A +
NW_005871066.1 3216149 3216149 C/T +
NW_005871073.1 753145 753145 C/A +
NW_005871075.1 5707870 5707870 T/A +
NW_005871076.1 3907289 3907289 G/A +
NW_005871082.1 6986210 6986210 A/T +
NW_005871090.1 7390034 7390034 A/G +
NW_005871097.1 3229647 3229647 G/A +
NW_005871100.1 365354 36

In [5]:
!tail -n 80 /local/home/sarahgw/WGS/SNPS/All_mylu/snps/outputVEP2

NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_006090548.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_006090549.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871123.1_4124241_A/G	NW_005871123.1:4124241	G	102421882	XM_014457399.2	Transcript	downstream_gene_variant	-	-	-	-	-	-	IMPACT=MODIFIER;DISTANCE=2309;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871126.1_2091046_C/T	NW_005871126.1:2091046	T	102433978	XM_006090652.3	Transcript	intron_variant	-	-	-	-	-	-	IMPACT=MODIFIER;STRAND=1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
NW_005871129.1_2603829_T/C	NW_005871129.1:2603829	C	102423892	XM_006090751.3	Transcript	intron_variant	-	-	-	-	-	-	IMPACT=MODIFIER;STRAND=-1;SOURCE=ref_Myoluc2.0_top_level_sorted.gff3.gz
